### ISSUE TITLE
Cannot concatenate boolean numpy arrays

### ISSUE THREAD

**Issue by leezu** [apache/incubator-mxnet#18213]
## Description
`mx.np.concatenate([mx.np.zeros((1,), dtype='bool')])` is broken.

### Error Message
```
File "src/operator/numpy/./np_matrix_op-inl.h", line 1160
MXNetError: Unknown type enum 7
```

cc @yzhliu

**Comment by leezu** [comment]
Related https://github.com/apache/incubator-mxnet/issues/16650

**Comment by Tommliu** [comment]
working on it

**Comment by yzhliu** [comment]
closed by #18277
